backend/src/layer/logic/mealplan_management/meal_plan_manager.rs

```diff
@@ -0,0 +1,103 @@
+    DataAccess: MealplanManagementDataAccess + Send + Sync,
+{
+    resolver: RelationResolver<DataAccess>,
+    parser: Parser,
+}
+
+impl<Parser, DataAccess> MealPlanManager<Parser, DataAccess>
+where
+    DataAccess: MealplanManagementDataAccess + Send + Sync,
+    Parser: MealplanParser + Send + Sync,
+{
+    pub const fn _new(database: DataAccess, meal_plan_parser: Parser) -> Self {
+        Self {
+            resolver: RelationResolver::_new(database),
+            parser: meal_plan_parser,
+        }
+    }
+
+    async fn start_resolving(&self, parse_canteens: Vec<ParseCanteen>, date: Date) {
+        for parse_canteen in parse_canteens {
+            let name = &parse_canteen.name.clone();
+            match self.resolver.resolve(parse_canteen, date).await {
+                Ok(_canteen) => debug!("resolved canteen '{name}' with no errors"),
+                Err(error) => warn!("resolved canteen '{name}' with errors: {error}"),
+            }
+        }
+    }
+}
+
+#[async_trait]
+impl<DataAccess, Parser> MensaParseScheduling for MealPlanManager<Parser, DataAccess>
+where
+    DataAccess: MealplanManagementDataAccess + Send + Sync,
+    Parser: MealplanParser + Send + Sync,
+{
+    //TODO transactions if relation_resolve fails and we dont want uncompleted meal plans?
+    /// This method starts the parsing procedure for all meal plans **of the current day**.<br>
+    /// After parsing, the raw data objects (`Vec<ParseCanteen>`) will be inserted by the `RelationResolver` with the current day.<br>
+    /// If during resolving an error occurs, the resolver stops and a log will be displayed.<br>
+    /// Each successful resolving process is also logged.
+    async fn start_update_parsing(&self) {
+        let today = Utc::now().date_naive();
+        match self.parser.parse(today).await {
+            Ok(parse_canteens) => {
+                self.start_resolving(parse_canteens, today).await;
+            }
+            Err(error) => {
+                warn!("canteens parsed with error and can't be resolved: {error}");
+            }
+        }
+    }
+
+    /// Similar to `start_update_parsing` this method starts the parsing procedure for all meal plans **for the next four weeks**.<br>
+    /// After parsing, the raw data objects (`Vec<(Date, Vec<ParseCanteen>>`) will be inserted by the `RelationResolver`.<br>
+    /// If during resolving an error occurs, the resolver stops and a log will be displayed.<br>
+    /// Each successful resolving process is also logged.
+    async fn start_full_parsing(&self) {
+        match self.parser.parse_all().await {
+            Ok(parse_tuples) => {
+                for (date, parse_canteens) in parse_tuples {
+                    self.start_resolving(parse_canteens, date).await;
+                }
+            }
+            Err(error) => {
+                warn!("canteens parsed with error and can't be resolved: {error}");
+            }
+        }
+    }
+}
+
```

Comment:
Tests are like production calls. They're not controlled in any way.

Reply:
I don't remind using .except as I'm not a fan of it. So I consider deleting this unit test section or use it for component test.

Reply:
The mock interfaces are not that interactive that we could expect uncommon behaviour... but it looks wrong without test in it...